### PR TITLE
[tag management] update docs links

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- update tag management docs links

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Verify that links work correctly (their placement is described in issues linked to this PR)

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/739
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/748

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.

### CODE REVIEW

File: ui/src/app/tags/components/DefinitionField/DefinitionField.tsx [hunk 1]
```diff
@@ -35,12 +35,13 @@ const getDefinitionError = (
   definitionErrorId: string
 ) => {
   if (errors.definition?.includes(INVALID_XPATH_ERROR)) {
-    // TODO: Add the link to the docs:
-    // https://github.com/canonical-web-and-design/app-tribe/issues/748
     return (
       <span id={definitionErrorId}>
         The definition is an invalid XPath expression. See our{" "}
-        <a href="#todo">XPath documentation</a> for more examples.
+        <a href="https://maas.io/docs/how-to-work-with-tags#heading--xpath-expressions">
```
Comment (anthonydillon) [inline ui/src/app/tags/components/DefinitionField/DefinitionField.tsx:41]: There is no heading--xpath-expressions id on this page

Comment (anthonydillon) [reply ui/src/app/tags/components/DefinitionField/DefinitionField.tsx:41]: There are no examples of valid and invalid XPath expressions in these docs.

File: ui/src/app/tags/components/DefinitionField/DefinitionField.tsx [hunk 2]
```diff
@@ -85,12 +86,14 @@ export const DefinitionField = ({ id }: Props): JSX.Element => {
...
         rows={3}
       />
-      {/* // TODO: Add the link to the docs:
-          // https://github.com/canonical-web-and-design/app-tribe/issues/748 */}
       <p className="p-form-help-text u-sv1">
         Add an XPath expression as a definition. MAAS will auto-assign this tag
         to all current and future machines that match this definition. More
-        about how to use <a href="#todo">XPath Expression</a>.
+        about how to use{" "}
+        <a href="https://maas.io/docs/how-to-work-with-tags#heading--xpath-expressions">
```
Comment (anthonydillon) [inline ui/src/app/tags/components/DefinitionField/DefinitionField.tsx:93]: Same here. Also, the link should be something like "Learn more about..." not just More about.

File: ui/src/app/tags/components/DefinitionField/DefinitionField.tsx [hunk 3]
```diff
@@ -85,12 +86,14 @@ export const DefinitionField = ({ id }: Props): JSX.Element => {
...
+          XPath Expression
```
Comment (anthonydillon) [inline ui/src/app/tags/components/DefinitionField/DefinitionField.tsx:94]: This is not a name, it should be "XPath expression"

File: ui/src/app/tags/components/KernelOptionsField/KernelOptionsField.tsx [hunk 4]
```diff
@@ -61,12 +61,13 @@ export const KernelOptionsField = ({
...
-        // TODO: Add the link to the docs:
-        // https://github.com/canonical-web-and-design/app-tribe/issues/748
         <>
           Kernel options are appended to the kernel command line during booting
           while machines are commissioning or deploying. Read more about kernel
-          options in <a href="#todo">tag management.</a>
+          options in{" "}
+          <a href="https://maas.io/docs/how-to-work-with-tags#heading--kernel-options">
+            tag management.
+          </a>
```
Comment (anthonydillon) [inline ui/src/app/tags/components/KernelOptionsField/KernelOptionsField.tsx:70]: Can we extend this link around the sentence starting "Read more ...."

File: ui/src/app/tags/components/KernelOptionsField/KernelOptionsField.tsx [hunk 5]
```diff
@@ -61,12 +61,13 @@ export const KernelOptionsField = ({
...
       component={Textarea}
       help={
-        // TODO: Add the link to the docs:
-        // https://github.com/canonical-web-and-design/app-tribe/issues/748
         <>
           Kernel options are appended to the kernel command line during booting
           while machines are commissioning or deploying. Read more about kernel
-          options in <a href="#todo">tag management.</a>
+          options in{" "}
+          <a href="https://maas.io/docs/how-to-work-with-tags#heading--kernel-options">
```
Comment (anthonydillon) [inline ui/src/app/tags/components/KernelOptionsField/KernelOptionsField.tsx:68]: No heading reference in the document

File: ui/src/app/tags/views/TagList/TagTable/TagTable.tsx [hunk 6]
```diff
@@ -224,7 +222,7 @@ const TagTable = ({
                         )
                       )}
                       <br />
-                      <a href="#todo">
+                      <a href="https://maas.io/docs/how-to-work-with-tags#heading--automatic-tags">
```
Comment (anthonydillon) [inline ui/src/app/tags/views/TagList/TagTable/TagTable.tsx:225]: Missing heading reference

File: ui/src/app/tags/views/TagList/TagTable/TagTable.tsx [hunk 7]
```diff
@@ -224,7 +222,7 @@ const TagTable = ({
...
                         Check the documentation about automatic tags.
```
Comment (anthonydillon) [inline ui/src/app/tags/views/TagList/TagTable/TagTable.tsx:226]: This one is the entire sentence :+1:

File: ui/src/app/tags/views/TagUpdate/TagUpdateFormFields/TagUpdateFormFields.tsx [hunk 8]
```diff
@@ -65,14 +65,15 @@ export const TagUpdateFormFields = ({ id }: Props): JSX.Element | null => {
...
         <Col size={6}>
-          {/* // TODO: Add the link to the docs:
-          // https://github.com/canonical-web-and-design/app-tribe/issues/748 */}
           {!tag.definition ? (
             <Definition label={Label.Definition}>
               <span className="p-form-help-text">
                 This is a manual tag. Definitions cannot be added to manual
                 tags. To learn more about this, check our{" "}
-                <a href="#todo">XPath documentation</a>.
+                <a href="https://maas.io/docs/how-to-work-with-tags#heading--xpath-expressions">
```
Comment (anthonydillon) [inline ui/src/app/tags/views/TagUpdate/TagUpdateFormFields/TagUpdateFormFields.tsx:73]: Same here:
- Id missing from the document
- Can we make the link the entire sentence for a11y and provide a clearer user expectation when clicking this link?
- Should these links open in a new tab?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few comments in the code.

File: ui/src/app/tags/views/TagList/TagTable/TagTable.tsx (hunk 6 above)
Comment (petermakowski) [reply ui/src/app/tags/views/TagList/TagTable/TagTable.tsx:225]: > Missing heading reference

Sorry, these pages haven't been updated yet, didn't expect you to jump on this so quickly. 😄  @amylily1011 do you know what's the ETA on this?

File: ui/src/app/tags/views/TagUpdate/TagUpdateFormFields/TagUpdateFormFields.tsx (hunk 8 above)
Comment (petermakowski) [reply ui/src/app/tags/views/TagUpdate/TagUpdateFormFields/TagUpdateFormFields.tsx:73]: Couldn't agree more on your points, but I didn't even look at the context as I was making quick link updates for @huwshimi - we need this before the 3.2 release today.

File: ui/src/app/tags/views/TagList/TagTable/TagTable.tsx (hunk 6 above)
Comment (amylily1011) [reply ui/src/app/tags/views/TagList/TagTable/TagTable.tsx:225]: I got all the screenshots now, will go through the content with Bill tonight. @anthonydillon Can we skip this part for now?

File: ui/src/app/tags/components/DefinitionField/DefinitionField.tsx (hunk 1 above)
Comment (amylily1011) [reply ui/src/app/tags/components/DefinitionField/DefinitionField.tsx:41]: Can we skip the content part for now? I created another [ticket](https://github.com/canonical-web-and-design/app-tribe/issues/807) to QA Doc content and dead links.

Comment (anthonydillon) [reply ui/src/app/tags/components/DefinitionField/DefinitionField.tsx:41]: Ok, seems strange to QA something that is broken but as long as the content is published and tested before the release then :+1:

File: ui/src/app/tags/views/TagUpdate/TagUpdateFormFields/TagUpdateFormFields.tsx (hunk 8 above)
Comment (anthonydillon) [reply ui/src/app/tags/views/TagUpdate/TagUpdateFormFields/TagUpdateFormFields.tsx:73]: I think the main part here which shouldn't be contentious is to improve the link text.

Comment (petermakowski) [reply ui/src/app/tags/views/TagUpdate/TagUpdateFormFields/TagUpdateFormFields.tsx:73]: When it comes to opening docs in a new tab - I'm pretty sure we decided against it as a general rule but can't find a source at the moment. Maybe @amylily1011 could help?

**Comment by petermakowski** [comment]
@anthonydillon  I made the changes you requested, could you please look again.

**Review by anthonydillon — APPROVED** [review]
LGTM thanks for the changes

**Pull request merged**